### ISSUE TITLE
Questions regarding shift

### ISSUE THREAD

**Issue by ghost** [proepkes/SpeedDate#1]
Two questions,
Firstly, What were the problems/reasons which led to a LiteNetLib implementation rather than the Darkrift one?
Secondly, which version of LiteNetLib is SpeedDate using?

**Comment by proepkes** [comment]
I've moved away from Darkrift for 2 reasons:

1: closed source. If there is a bug in the network-library (and there are a [few](https://github.com/DarkRiftNetworking/DarkRift-Networking/issues)) there is no (legal) way to get it fixed other than waiting for the author to fix it. Also the development seems rather slow to me.

2: Cluttering with MessageTags. A MasterServer uses lots of query-response-messages so the DarkRifts 'fire-and-forget'-API leads to unnecessary overhead for a simple response-message (messageTag + successMessageTag + errorMessageTag). Instead of waiting for a possible implementation [here](https://github.com/DarkRiftNetworking/DarkRift-Networking/issues/11) I've decided to bring SpeedDate to life. I _could've_ build a message-id-system on top of DarkRift, but this would re-invent the wheel because MSF already has a working implementation.

Using LiteNetLib was a personal preference. It is actively developed, open source and has a nice API. SpeedDate uses LiteNetLib's master-branch (commit [c8cb2b7](https://github.com/RevenantX/LiteNetLib/commit/c8cb2b7206c212f15f1cf53839eea12efeccffbd) as the time of writing). 

_In my opinion_: DarkRift is suited for "GameServer -> Client"-communication but lacks features for Server-to-Server- or Query-Response-communication
